### PR TITLE
Add section on workflows and mlflow-builder workflow extension

### PR DESCRIPTION
**Pull request by stefannica**
Preview available:
- here: https://stefannica.github.io/docs/dev/workflows/workflows/
- and here: https://stefannica.github.io/docs/dev/workflows/mlflow-builder/

### CODE REVIEW

File: docs/workflows/workflows.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    default: auto
+outputs:
+  - name: prediction-url
+    description: "The URL where the exposed prediction service endpoint can be contacted to run predictions."
+    type: string
+```
+
+This is where the workflow's global parameters, inputs and outputs are declared. Currently, the only type of supported input for FuseML workflows and steps is the codeset. In this example:
+
+- the `mlflow-codeset` input is a codeset that must contain an MLflow project with python code that will be used to train a ML model
```
Comment (jsuchome) [inline docs/workflows/workflows.md:104]: It might be a good idea to add a link to the description of codesets

**Review by jsuchome — DISMISSED** [review]
Nice

File: docs/workflows/workflows.md (hunk 1 above)
Comment (stefannica) [reply docs/workflows/workflows.md]: done (I hope)

File: docs/workflows/kserve-predictor.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+- scikit-learn pickled models
+- TensorFlow (saved_model) models
+- ONNX models
+
+The KServe predictor step expects a model URL to be supplied as input, pointing to the location in an MLflow artifact store where the model is stored. Currently, S3 is the only protocol supported for the MLflow artifact store back-end.
+
+The predictor performs the following tasks:
+
+- downloads the model locally from the MLflow artifact store
+- if so instructed, it auto-detects the model format based on the information stored in the MLflow artifact store and decides which KServe predictor engine to use for it. Otherwise, it validates the model format against the type of predictor engine specified as input.
```
Comment (jsuchome) [inline docs/workflows/kserve-predictor.md:16]: Maybe you can write explicitly the rules for such selections

File: docs/workflows/kserve-predictor.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+- it performs some minor conversion tasks required to adapt the input MLflow model directory layout to the one required by KServe
+- it creates a KServe prediction service to serve the model
+- finally, it registers the KServe prediction service with FuseML as an Application object. Information about the Application, such as the type and exposed inference URL can be retrieved at any time using the FuseML API and CLI.
+
+## Using the KServe Predictor Step
+
+TBD
```
Comment (jsuchome) [inline docs/workflows/kserve-predictor.md:23]: Do you expect to write more here?

File: docs/workflows/workflows.md (hunk 1 above)
Comment (jsuchome) [reply docs/workflows/workflows.md]: Hm, I do not see any reference there. I had in mind a reference to codeset definition, but maybe we do not have one...

File: docs/workflows/mlflow-builder.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+- `mlflow_artifact_path` - sub-path under the MLflow artifact repository that should be returned as output. By default, the `model` sub-directory is returned, because this is where MLflow stores the ML models.
+- `verbose` - set to `true` to enable verbose logging in the MLflow runtime step (default is `false`).
+
+The MLflow runtime workflow step can also take in additional environment variables that are passed transparently to the MLflow runtime and used to configure how the MLflow runtime can access a remote MLflow tracking server and artifact store. For more information on what variables are available and how they can be used, please refer to the [relevant section in the official MLflow documentation](https://www.mlflow.org/docs/latest/tracking.html#logging-to-a-tracking-server). Only a subset of the available variables are listed here:
+
+> :warning: Note that some of these environment variables contain sensitive data, such as keys and passwords and should not be explicitly configured as workflow step env vars. Instead, they should be configured in the FuseML Extension Registry and only referenced in the FuseML workflows as extension requirements.
```
Comment (jsuchome) [inline docs/workflows/mlflow-builder.md:92]: This is rendered strangely, I assume you wanted something else...

File: docs/workflows/workflows.md (hunk 1 above)
Comment (stefannica) [reply docs/workflows/workflows.md]: no, we don't have a section dedicated to codesets :-(

File: docs/workflows/kserve-predictor.md (hunk 2 above)
Comment (stefannica) [reply docs/workflows/kserve-predictor.md:16]: Totally agree, but I'll have to do it in a follow-up PR. This is just a draft I had to include in this PR to silence the CI.

File: docs/workflows/kserve-predictor.md (hunk 3 above)
Comment (stefannica) [reply docs/workflows/kserve-predictor.md:23]: Yes. I'll add more documentation around remaining extensions in a follow-up PR.

File: docs/workflows/mlflow-builder.md (hunk 4 above)
Comment (stefannica) [reply docs/workflows/mlflow-builder.md]: you're right, thanks for noticing. This format works with the github markdown renderer, but it doesn't work here. I fixed it to look like a note section.

**Review by jsuchome — APPROVED** [review]

**Pull request merged**